src/main/java/org/mockito/internal/matchers/Equals.java

```diff
@@ -51,12 +51,18 @@ public int hashCode() {
     }
 
     @Override
-    public String toStringWithType() {
-        return "(" + wanted.getClass().getSimpleName() + ") " + describe(wanted);
+    public String toStringWithType(boolean useFullyQualifiedClassName) {
+        return "(" + (useFullyQualifiedClassName ? wanted.getClass().getCanonicalName() : wanted.getClass().getSimpleName()) + ") " + describe(wanted);
     }
 
     @Override
     public boolean typeMatches(Object target) {
         return wanted != null && target != null && target.getClass() == wanted.getClass();
     }
+
+    @Override
+    public Class getWantedClass() {
```

Comment:
We already define `getWanted` above, but it is `protected`. I think we should make that `public` instead and then remove this function. WDYT?

Reply:
Had thought the same but was not sure whether to make it public . Also have added 3 test cases. I tried finding classes with same name through java library, but if two classes had same name, they didn't have a toString implemented. 
So i have made a dummy class named Date in IMethods. 
Also let me know if the method names are too long.